### PR TITLE
Don't validate affected rows for insert operations

### LINKED ISSUE #2131: Update: SQL Server: Selecting @@ROWCOUNT after INSERT
Inserts generally either succeed, or an exception is raised. Do we need to select back @@ROWCOUNT?

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Issue SET NOCOUNT ON on all operations

Fixes #2131

### CODE REVIEW

File: src/EntityFramework.MicrosoftSqlServer/Update/Internal/ISqlServerUpdateSqlGenerator.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Data.Entity.Update.Internal
 {
     public interface ISqlServerUpdateSqlGenerator : IUpdateSqlGenerator
     {
-        ResultsGrouping AppendBulkInsertOperation(
+        ResultsetMapping AppendBulkInsertOperation(
```
Comment (Artur2) [inline src/EntityFramework.MicrosoftSqlServer/Update/Internal/ISqlServerUpdateSqlGenerator.cs]: Maybe ResultSetMapping

File: src/EntityFramework.MicrosoftSqlServer/Update/Internal/SqlServerUpdateSqlGenerator.cs [hunk 2]
```diff
@@ -22,8 +22,8 @@ public class SqlServerUpdateSqlGenerator : UpdateSqlGenerator, ISqlServerUpdateS
         {
             _typeMapper = typeMapper;
         }
-        
-        public virtual ResultsGrouping AppendBulkInsertOperation(
+
+        public virtual ResultsetMapping AppendBulkInsertOperation(
```
Comment (Artur2) [inline src/EntityFramework.MicrosoftSqlServer/Update/Internal/SqlServerUpdateSqlGenerator.cs]: Maybe ResultSetMapping ?

File: src/EntityFramework.MicrosoftSqlServer/Update/Internal/SqlServerUpdateSqlGenerator.cs [hunk 3]
```diff
@@ -82,19 +82,18 @@ public class SqlServerUpdateSqlGenerator : UpdateSqlGenerator, ISqlServerUpdateS
...
-                : ResultsGrouping.OneResultSet;
+            return resultSetCreated ?
+                defaultValuesOnly
+                    ? ResultsetMapping.LastInResultSet
+                    : ResultsetMapping.NotLastInResultSet
+                : ResultsetMapping.NoResultSet;
         }
 
-        public override void AppendUpdateOperation(StringBuilder commandStringBuilder, ModificationCommand command, int commandPosition)
+        public override ResultsetMapping AppendUpdateOperation(StringBuilder commandStringBuilder, ModificationCommand command, int commandPosition)
```
Comment (Artur2) [inline src/EntityFramework.MicrosoftSqlServer/Update/Internal/SqlServerUpdateSqlGenerator.cs]: ResultSetMapping ?

**Comment by natemcmaster** [comment]
Looks like some SQlite tests are failing because they asser `SELECT changes()` is included in the INSERT operation. I assume this means we just need to update the tests...

**Comment by anpete** [comment]
:shipit:

**Pull request merged**